### PR TITLE
Create and test fromHexString and fromBytes for SszBitlist

### PR DESCRIPTION
**Pull request by oceans404**
## PR Description

this PR adds 2 methods for working with SszBitlist

- fromBytes: Create an SszBitlist directly from bytes
- fromHexString: Create an SszBitlist from a hex string (wrapper around fromBytes)

These methods are now available on SszBitlistSchema, so you can do

```
SszBitlist myBitlist = schema.fromBytes(someBytes);
SszBitlist myOtherBitlist = schema.fromHexString("0x01020304");
```

## Fixed Issue(s)

Issue #8505

### CODE REVIEW

**Comment by oceans404** [comment]
hey @rolfyone! is it possible to rerun ci/circleci: acceptanceTests on this pr?

File: infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/collections/SszBitlistTest.java [hunk 1]
```diff
@@ -394,4 +402,61 @@ void testBitEmptyListSsz(final SszBitlist bitlist) {
...
+    SszBitlist deserialized = bitlist.getSchema().fromHexString(hexString);
+
+    assertThat(deserialized).isEqualTo(bitlist);
+    assertThat(deserialized.size()).isEqualTo(bitlist.size());
+    assertThatIntCollection(deserialized.getAllSetBits()).isEqualTo(bitlist.getAllSetBits());
+    assertThat(deserialized.hashTreeRoot()).isEqualTo(bitlist.hashTreeRoot());
+  }
+
+  @Test
+  public void testFromHexStringEdgeCases() {
```
Comment (gfukushima) [inline infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/collections/SszBitlistTest.java]: I think this would be even better if we break this general edge cases into individual tests?

File: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/impl/SszBitlistSchemaImpl.java [hunk 2]
```diff
@@ -154,4 +154,13 @@ public int flushWithBoundaryBit(final SszWriter writer, final int boundaryBitOff
       }
     }
   }
+
+  @Override
+  public SszBitlist fromBytes(final Bytes bytes) {
+    checkArgument(bytes != null, "Input bytes cannot be null");
+    try (SszReader reader = SszReader.fromBytes(bytes)) {
+      TreeNode node = sszDeserializeTree(reader);
```
Comment (gfukushima) [inline infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/impl/SszBitlistSchemaImpl.java]: nit: this can use the `final` modifier

**Review by gfukushima — COMMENTED** [review]
minor suggestion for the general edge case tests created. other than that I think it looks good!

File: infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/collections/SszBitlistTest.java (hunk 1 above)
Comment (oceans404) [reply infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/collections/SszBitlistTest.java]: good point - updated!

File: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/impl/SszBitlistSchemaImpl.java (hunk 2 above)
Comment (oceans404) [reply infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/impl/SszBitlistSchemaImpl.java]: great catch. updated!

**Comment by oceans404** [comment]
> minor suggestion for the general edge case tests created. other than that I think it looks good!

thanks! updated based on your suggestions. can you take another look when you have a moment, @gfukushima?

**Review by gfukushima — APPROVED** [review]
lgtm, thanks for the contribution!

**Comment by rolfyone** [comment]
Thanks for the contribution @oceans404, we'll merge this soon!
